frame/nomination-pools/src/lib.rs

```diff
@@ -639,66 +739,51 @@ impl<T: Config> BondedPool<T> {
 			},
 			// Any delegator who is not the depositor can always unbond themselves
 			(true, false) => (),
-			// The depositor can only start unbonding if the pool is already being destroyed and
-			// they are the delegator in the pool. Note that an invariant is once the pool is
-			// destroying it cannot switch states, so by being in destroying we are guaranteed no
-			// other delegators can possibly join.
 			(_, true) => {
-				ensure!(target_delegator.points == self.points, Error::<T>::NotOnlyDelegator);
-				ensure!(self.is_destroying(), Error::<T>::NotDestroying);
+				if self.is_destroying_and_only_bonded_delegator(target_delegator.active_points()) {
+					// if the pool is about to be destroyed, anyone can unbond the depositor, and
+					// they can fully unbond.
+				} else {
+					// only the depositor can partially unbond, and they can only unbond up to the
+					// threshold.
+					ensure!(is_permissioned, Error::<T>::DoesNotHavePermission);
+					let new_depositor_points =
+						target_delegator.active_points().saturating_sub(unbonding_points);
+					ensure!(
+						new_depositor_points >= MinCreateBond::<T>::get(),
```

Comment:
IMO `MinCreateBond` should represent a balance, not points. If we take that `MinCreatedBond` is a minimum balance to bond, then this seems to make the assumption that points should be 1:1 with balance, however this could easily be off. For example, say there was a slash and the depositor's balance went well below `MinCreateBond`. Then they bond a little bit more funds, but not enough to meet `MinCreateBond`, however since the points will grow at a much higher rate then the balance, there points could be significantly greater then `MinCreateBond`, when they don't even have `MinCreateBond` amount bonded

Reply:
yeah, I was totally wrong here and I see the issue. 

Not super clean, but cb56178606de72ec6649c6bee5cd2b7df2df28e3 should fix it.

Reply:
since now we use the balance to point conversion in places where we actually don't intend to unbond, I also did this refactor in a separate commit so we can discuss it. 011e8bc50a5876c458356d345b6d11ee4c824295